src/panels/home/ha-panel-home.ts

```
@@ -0,0 +1,144 @@
+
+  @property({ attribute: false }) public route?: Route;
+
+  @property({ attribute: false }) public panel?: PanelInfo;
+
+  @state() private _lovelace?: Lovelace;
+
+  public willUpdate(changedProps: PropertyValues) {
+    super.willUpdate(changedProps);
+    // Initial setup
+    if (!this.hasUpdated) {
+      this.hass.loadFragmentTranslation("lovelace");
+      this._setLovelace();
+      return;
+    }
+
+    if (!changedProps.has("hass")) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as this["hass"];
+    if (oldHass && oldHass.localize !== this.hass.localize) {
+      this._setLovelace();
+      return;
+    }
+
+    if (oldHass && this.hass) {
+      // If the entity registry changed, ask the user if they want to refresh the config
+      if (
+        oldHass.entities !== this.hass.entities ||
+        oldHass.devices !== this.hass.devices ||
+        oldHass.areas !== this.hass.areas ||
+        oldHass.floors !== this.hass.floors
+      ) {
+        if (this.hass.config.state === "RUNNING") {
+          this._debounceRegistriesChanged();
+          return;
+        }
+      }
+      // If ha started, refresh the config
+      if (
+        this.hass.config.state === "RUNNING" &&
+        oldHass.config.state !== "RUNNING"
+      ) {
+        this._setLovelace();
+      }
+    }
+  }
+
+  private _debounceRegistriesChanged = debounce(
+    () => this._registriesChanged(),
+    200
+  );
+
+  private _registriesChanged = async () => {
+    this._setLovelace();
+  };
+
+  protected render() {
+    if (!this._lovelace) {
+      return nothing;
+    }
+
+    return html`
+      <hui-root
+        .hass=${this.hass}
+        .narrow=${this.narrow}
+        .lovelace=${this._lovelace}
+        .route=${this.route}
+        .panel=${this.panel}
+      ></hui-root>
+    `;
+  }
+
+  private async _setLovelace() {
+    const config = await generateLovelaceDashboardStrategy(
+      HOME_LOVELACE_CONFIG,
+      this.hass
+    );
```

There should be some error handling here or it will fail silently. I had this issue on the energy dashboard somewhere

---

All the strategies fail silently, we should fix that. I will do error handling in a future PR.